.github/workflows/build.yml

```
@@ -0,0 +1,40 @@
+name: Earthly +build
+
+on:
+  merge_group:
+  pull_request:
+  push:
+    branches:
+      - main
+
+env:
+  FORCE_COLOR: 1
+  CARGO_TERM_COLOR: always
+  NIGHTLY_TOOLCHAIN: nightly
+
+jobs:
+  build:
+    timeout-minutes: 30
+    runs-on: ubuntu-latest
+    env:
+      FORCE_COLOR: 1
+      DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+      DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+
+    steps:
+      - uses: earthly/actions-setup@v1
+        with:
+          use-cache: true
+          version: v0.8.0
+
+      # Pulling rust toolchain & install linux dependencies
+      - uses: dtolnay/rust-toolchain@stable
+
+      # Setup repo and add caching
+      - uses: actions/checkout@v4
+
+      - name: Docker Login
+        run: docker login ghcr.io -u $DOCKERHUB_USERNAME -p $DOCKERHUB_TOKEN
```

```suggestion
        run: docker login ghcr.io -u ${{ github.actor }} -p ${{ secrets.GITHUB_TOKEN }}
```

Then you can remove the `DOCKERHUB` env vars